src/ir_visitor/visitor.rs

```diff
@@ -29,16 +32,30 @@ impl Action {
     }
 }
 
+/// Contains information passed to visitor functions
+pub struct VisitorData<'comp> {
```

Comment:
Is the lifetime name correct? I thought the lifetime is that of the entire pass and not just the component?

Reply:
Oh, I guess you have to change the context after traversing each component. Maybe say that in the comment?